flink-table/flink-table-common/src/test/java/org/apache/flink/table/types/inference/InputTypeStrategiesTestBase.java

```diff
@@ -45,39 +44,49 @@
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
 /** Base class for testing {@link InputTypeStrategy}. */
-@RunWith(Parameterized.class)
 public abstract class InputTypeStrategiesTestBase {
 
-    @Parameterized.Parameter public TestSpec testSpec;
-
-    @Test
-    public void testStrategy() {
+    @ParameterizedTest(name = "{index}: {0}")
+    @MethodSource(
+            value = {
+                "org.apache.flink.table.types.inference.ComparableInputTypeStrategyTest#testData",
+                "org.apache.flink.table.types.inference.strategies.CurrentWatermarkInputTypeStrategyTest#testData",
+                "org.apache.flink.table.types.inference.InputTypeStrategiesTest#testData",
+                "org.apache.flink.table.types.inference.strategies.RepeatingSequenceInputTypeStrategyTest#testData",
+                "org.apache.flink.table.types.inference.SubsequenceInputTypeStrategyTest#testData",
+                "org.apache.flink.table.types.inference.strategies.SymbolArgumentTypeStrategyTest#testData",
+                "org.apache.flink.table.types.inference.strategies.TypeLiteralArgumentTypeStrategyTest#testData"
+            })
```

Comment:
It leads to cartesian of tests and input....
need to think how to cope with this